api/envoy/api/v2/auth/cert.proto

```diff
@@ -127,7 +127,14 @@ message CertificateValidationContext {
   //
   // See :ref:`the TLS overview <arch_overview_ssl_enabling_verification>` for a list of common
   // system CA locations.
-  core.DataSource trusted_ca = 1;
+  oneof trusted_ca_type {
```

Comment:
This nested structure here adds more complexity, can we somehow guarantee that there is no SDS to SDS dependency at API level? (i.e. SDS delivering CertificateValidationContext with Trust CA from SDS).